pandas/tests/test_frame.py

```diff
@@ -14751,6 +14751,16 @@ def test_query_builtin(self):
         result = df.query('sin > 5', engine=engine, parser=parser)
         tm.assert_frame_equal(expected, result)
 
+    def test_query_with_assign_statement(self):
+        df = DataFrame({'a': [1, 2, 3], 'b': ['a', 'b', 'c']})
+        a_before = df['a'].copy()
+        self.assertRaisesRegexp(
+            NotImplementedError, "'Assign' nodes are not implemented",
```

Comment:
`NotImplementedError` is what you currently get when you do unsupported operations within a query/eval and hit an unsupported node, e.g.:

```python
import pandas as pd
df = pd.DataFrame({'a': [1, 2, 3], 'b': ['a', 'b', 'c']})
df.eval("{'x': 1}")
NotImplementedError 
/home/me/.local/lib/python2.7/site-packages/pandas/computation/expr.pyc in visit(self, node, **kwargs)
    312         method = 'visit_' + node.__class__.__name__
    313         visitor = getattr(self, method)
--> 314         return visitor(node, **kwargs)
    315 
    316     def visit_Module(self, node, **kwargs):

/home/me/.local/lib/python2.7/site-packages/pandas/computation/expr.pyc in f(self, *args, **kwargs)
    203     def f(self, *args, **kwargs):
    204         raise NotImplementedError("{0!r} nodes are not "
--> 205                                   "implemented".format(node_name))
    206     return f
    207 
NotImplementedError: 'Dict' nodes are not implemented
```

So at the moment it's just working like any other invalid query/eval. For this specific case I'll probably try to catch the exception within `DataFrame.query()` and spit out a more meaningful
message. Not sure where you'd have to intercept the exception if you wanted to raise more
meaningful error messages from `eval` more generally.